velox/common/base/BitUtil.h

```diff
@@ -691,8 +691,19 @@ bool inline hasIntersection(
       });
 }
 
-inline int32_t countLeadingZeros(uint64_t word) {
-  return __builtin_clzll(word);
+template <typename T = uint64_t>
+inline int32_t countLeadingZeros(T word) {
+  static_assert(std::is_same_v<T, uint64_t> || std::is_same_v<T, __uint128_t>);
+  if (word == 0) {
+    return sizeof(T) * 8;
+  }
+  if constexpr (std::is_same_v<T, uint64_t>) {
+    return __builtin_clzll(word);
+  } else {
+    uint64_t hi = word >> 64;
+    uint64_t lo = static_cast<uint64_t>(word);
+    return (hi == 0) ? 64 + __builtin_clzll(lo) : __builtin_clzll(hi);
```

Comment:
Can lo be zero? Do we need to check for that as well? Can you make sure we have a test case for that?

Reply:
lo can be zero, if go to this branch `(hi == 0) ? 64 + __builtin_clzll(lo) `, high is 0 and lo is 0, the word should be 0, test have covered by `EXPECT_EQ(countLeadingZeros<__uint128_t>(0), 128)`;